torchgeo/datasets/satlas.py

```diff
@@ -0,0 +1,700 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""SatlasPretrain dataset."""
+
+import glob
+import json
+import os
+from collections.abc import Callable, Iterable
+from typing import ClassVar, TypedDict
+
+import numpy as np
+import torch
+from matplotlib import pyplot as plt
+from matplotlib.figure import Figure
+from PIL import Image
+from torch import Tensor
+
+from .errors import DatasetNotFoundError
+from .geo import NonGeoDataset
+from .utils import Path, download_and_extract_archive, extract_archive
+
+
+class _Task(TypedDict, total=False):
+    BackgroundInvalid: bool
+    categories: list[str]
+    colors: list[list[int]]
+    type: str
+
+
+# https://github.com/allenai/satlas/blob/main/satlas/model/dataset.py
+TASKS: dict[str, _Task] = {
```

Comment:
Are y'all comfortable with us copy-n-pasting this dict from your code without license attribution? Not sure how this works from a legal perspective. So far we're only using the colors, but we may also want to use the categories. We could reformat this if needed.

Reply:
Yes that is fine with us for sure.